### PR TITLE
Make the test suite detect bugs

### PR DESCRIPTION
**Pull request by javierm**
## References

In order to make the customizations easier, we've brought the following changes from CONSUL:

* Pull request consul/consul#4707 (these tests were almost always failing on this repo)
* Pull request consul/consul#4715
* Pull request consul/consul#4755

## Objectives

* Make sure the test suite detects bugs when we introduce them

### CODE REVIEW

File: app/models/custom/verification/residence.rb [hunk 1]
```diff
@@ -56,6 +56,10 @@ def document_number_format
     end
   end
 
+  def local_postal_code
+    errors.add(:postal_code, I18n.t("verification.residence.new.error_not_allowed_postal_code")) unless valid_postal_code?
```
[the remark this answers is not in the document]
Comment (javierm) [reply app/models/custom/verification/residence.rb:60]: This line actually comes from CONSUL, so keeping it like this for now.

**Review by Senen — COMMENTED** [review]
Hi @javierm, here is the initial review.

**07db690c4f Include Romanian language in the test environment**
I think, the second part of the message is no longer needed 🤔. 

**a8904a700d Disable specs testing investment supports**
In this commit, we mentioned the commit 6906d4c47, where the custom model for Budget::Investment was added copying the whole original file. I've made a diff between files and the only difference is:

```
    def should_show_votes?
+      false
-      budget.selecting?
    end

    def should_show_vote_count?
+      false
-      budget.valuating?
    end
```

It happens the same in the other custom models `Verification::Residence` , `Verification::Sms` and `Abilities::Common`. I've added a commit removing the duped code from custom classes, however, I was not able to apply the same to Verification classes as they modify validations.

Let me know what you think.

Later I'll take a deep look at changes introduced to the Verification process. I did not get it the first take.

**Comment by javierm** [comment]
> **[07db690](https://github.com/code4romania/consul/commit/07db690c4fa0b8c33887c965913d0ea348c20197) Include Romanian language in the test environment**
> I think, the second part of the message is no longer needed :thinking:

@Senen Thanks! I've updated it. At first I thought we could remove French and Dutch but of course they're there because they're actually used in the test suite :relieved:.

**Comment by javierm** [comment]
Hi, @decembrya :smile:.

Here's a missing the translation to English: "Verificându-vă informațiile veți avea acces la procesul de bugetare participativă". In case it helps, the key for this translation is `verification.user_permission_info_budgets`.

We've noticed the other texts we mentioned which lack an English translation aren't available in Romanian either. The missing texts are shown on this screenshot; they appear as "User Information", "Field", "Value", "Email", "Document Number" and "Phone Number" but note they are **not** the texts in English but placeholders automatically generated by the application. In this case, we'd need both the English and Romanian texts:

The profile of a user, with placeholder texts instead of Romanian ones

**Comment by javierm** [comment]
@decembrya Sorry for not noticing earlier :pray:. We've just found the original Romanian texts were [accidentally removed in commit f86f48b16d](https://github.com/code4romania/consul/commit/f86f48b16d#diff-968ad7bcfe8dafadbb9f72b7ab07696e03415fc7b5c86dbd2bc93eca865adc85L896). Are these texts OK? If so, we'd only need the English translations:

```
      user_information: "Informații despre utilizator (vizibile doar pentru administratori)"
      field: Câmp
      value: Valoare
      email: Email
      document_number: CNP
      phone_number: Număr de telefon
```

**Comment by javierm** [comment]
@decembrya Regarding the help page, we thought that (as is usually the case) the link to the help page was removed as a way to disable the help page completely. But, since there's a chance that the people in Brașov would like to enable this page without linking it from the homepage, we're keeping the current behavior and the link to the help page won't be displayed even if the feature is enabled :ok_hand:. We've made changes so it doesn't affect tests checking the presence of a "Help" link on the homepage when the feature is enabled.

**Comment by decembrya** [comment]
Hello @javierm, the translations you found are correct in Romanian, here are the English equivalents. Let me know if there are other translations needed.

>       user_information: "Information about the user (visible only for administrators)"
>       field: Field
>       value: Value
>       email: Email
>       document_number: Document Number
>       phone_number: Phone number

**Comment by decembrya** [comment]
For the help page, please let me know if you need further action from my side on the current installation. 
As I understand, it the page is not currently displayed or accessible except via a direct link so disabling it should not be an issue.

**Comment by javierm** [comment]
@decembrya Thanks! Could you also translate "Verificându-vă informațiile veți avea acces la procesul de bugetare participativă"? Only this one left :tada:.

P.S. No action regarding the help page is required from your side.

**Comment by decembrya** [comment]
Oh, sorry I missed that one, here is the translation:
"By verifying your information you will gain access to the participatory budgeting process"

Thanks for clarifying on the help page  👍

**Review by taitus — APPROVED** [review]

**Pull request merged**